fastapi/openapi/utils.py

```diff
@@ -211,9 +211,13 @@ def get_openapi_path(
             operation.setdefault("responses", {}).setdefault(status_code, {})[
                 "description"
             ] = route.response_description
+            assert (
+                route.response_class and route.response_class.media_type
+            ), "A response class with media_type is needed to generate OpenAPI"
+            media_type: str = route.response_class.media_type
             operation.setdefault("responses", {}).setdefault(
                 status_code, {}
-            ).setdefault("content", {}).setdefault(route.response_class.media_type, {})[
```

Comment:
I think you'll be able to fix the mypy issue by using `response_class = route.response_class or route.default_response_class`. Or by setting `self.response_class: Type[Response] = response_class or default_response_class` in the route `__init__`.

Reply:
Oh nevermind, I see the problem, media_type itself could be None

Reply:
Thanks! I don't know I wasn't getting the same line locally, nor the error :thinking: 

I added an assertion that requires the `route.response_class` to exist. As these utils should only be called after adding the sub-routers to the app. So, by that point, they all should be some class (and not `None`).